extensions/amp-base-carousel/1.0/README.md

```diff
@@ -0,0 +1,731 @@
+  </style>
+</head>
+<bento-base-carousel id="my-carousel">
+  <div class="red-gradient"></div>
+  <div class="blue-gradient"></div>
+  <div class="green-gradient"></div>
+</bento-base-carousel>
+<div class="buttons" style="margin-top: 8px;">
+  <button id='prev-button'>Go to previous slide</button>
+  <button id='next-button'>Go to next slide</button>
+  <button id='go-to-button'>Go to slide with green gradient</button>
+</div>
+
+<script>
+  (async () => {
+    const carousel = document.querySelector('#my-carousel');
+    await customElements.whenDefined('bento-base-carousel');
+    const api = await carousel.getApi();
+
+    // programatically advance to next slide
+    api.next();
+
+    // set up button actions
+    document.querySelector('#prev-button').onclick = () => api.prev();
+    document.querySelector('#next-button').onclick = () => api.next();
+    document.querySelector('#go-to-button').onclick = () => api.goToSlide(2);
+  })();
+</script>
+```
+
+[/example]
+
+#### Interactivity and API usage
+
+Bento enabled components in standalone use are highly interactive through their API. The `bento-base-carousel` component API is accessible by including the following script tag in your document:
+
+```javascript
+await customElements.whenDefined('bento-base-carousel');
+const api = await carousel.getApi();
+```
+
+##### Actions
+
+The `bento-base-carousel` API allows you to perform the following actions:
+
+**next()**
+Moves the carousel forwards by `advance-count` slides.
+
+```javascript
+api.next();
+```
+
+**prev()**
+Moves the carousel backwards by `advance-count` slides.
+
+```javascript
+api.prev();
+```
+
+**goToSlide(index: number)**
+Moves the carousel to the slide specified by the `index` argument.
+Note: `index` will be normalized to a number greater than or equal to `0` and less than the number of slides given.
+
+```javascript
+api.goToSlide(0); // Advance to first slide.
+api.goToSlide(length - 1); // Advance to last slide.
+```
+
+##### Events
+
+The `bento-base-carousel` API allows you to register and respond to the following events:
+
+**slideChange**
+
+This event is triggered when the index displayed by the carousel has changed.
+The new index is available via `event.data.index`.
+
+```javascript
+carousel.addEventListener('slideChange', (e) => console.log(e.data.index))
+```
+
+#### Layout and style
+
+Each Bento component has a small CSS library you must include to guarantee proper loading without [content shifts](https://web.dev/cls/). Because of order-based specificity, you must manually ensure that stylesheets are included before any custom styles.
+
+```html
+<link rel="stylesheet" type="text/css" href="https://cdn.ampproject.org/v0/amp-base-carousel-1.0.css">
+```
+
+Alternatively, you may also make the light-weight pre-upgrade styles available inline:
+
+```html
+<style data-bento-boilerplate>
+  bento-base-carousel {
+    display: block;
+    overflow: hidden;
+    position: relative;
+  }
+</style>
+```
```

Comment:
I think we need to mention this information in the first paragraph as well, since it must be included.  Maybe:

You must include each Bento component's required CSS library to guarantee proper loading and before adding custom styles. Or use the light-weight pre-upgrade styles available inline. See Layout and style.

Reply:
Wait, is this untrue for P/React?

Reply:
For Preact, as long as they give explicit dimensions i.e. `width` and `height` on the component, they don't need this.

Reply:
Is the `You must include each Bento component's required CSS library to guarantee proper loading and before adding custom styles. Or use the light-weight pre-upgrade styles available inline. See Layout and style.` recommendation for the line to go under the `### Web component` header?

Reply:
Thanks for the clarification - moving it under the `### Web component` header is perfect